CDCgov/prime-simplereport
# Data model supports multiple providers and default provider

# BACKEND PULL REQUEST

## Related Issue

first of probably 5 PRs to implement #7982 

## Changes Proposed

- add concept of multiple providers to a facility
- add default ordering provider column, populate it with ordering provider data, and swap all references to ordering provider to instead get default provider or a provider from the facility's provider set
- when a new facility is created, the ordering provider is set as the default and added to the set of providers

## Additional Information
next PR will remove the ordering provider column (doing it in a separate merge to make rollbacks less risky)

next next PR will handle:
- backend will support adding, removing, listing providers and getting/updating the default provider
- update org initialization logic to fix how we're creating the default provider and facilities (e.g. see number of identical providers in your local DB)

## Testing

DB migration should populate `facility_providers` table and new `default_ordering_provider_id` column on `facility` table

test that existing provider related behavior is not impacted:
- create facility with a provider
- update provider
- delete a facility (provider is not marked as deleted)
